syntax = "proto3";

///////////////////////////////////////////////////////////////////////////////
// Membership
///////////////////////////////////////////////////////////////////////////////

enum Status {
    NONE = 0;
    ALIVE = 1;
    SUSPICIOUS = 2;
    CONFIRM = 3;
}

message Member {
    string address = 1;
    int32 port = 2;
    int64 version_number = 3;
    int64 heartbeat_counter = 4;
    int64 incarnation_number = 5;
    Status status = 6;
}

message Heartbeat {
    repeated Member members = 1;
}


///////////////////////////////////////////////////////////////////////////////
// Simple Distributed File System
///////////////////////////////////////////////////////////////////////////////

message SdfsMessage {
    int64 type = 1;
    bytes data = 2;
    bytes padding = 3;
}

///////////////////////////////////////////////////////////////////////////////
// Maple Juice
///////////////////////////////////////////////////////////////////////////////

message MapleJuiceMessage {
    int64 type = 1;
    bytes data = 2;
    bytes padding = 3;
}
